mastodon.go

```diff
@@ -232,14 +232,23 @@ func (c *Client) authenticate(ctx context.Context, params url.Values) error {
 	return nil
 }
 
+// Convenience constants for Toot.Visibility
+const (
+	VisibilityPublic        = "public"
+	VisibilityUnlisted      = "unlisted"
+	VisibilityFollowersOnly = "private"
+	VisibilityDirectMessage = "direct"
+)
+
 // Toot is struct to post status.
 type Toot struct {
-	Status      string `json:"status"`
-	InReplyToID ID     `json:"in_reply_to_id"`
-	MediaIDs    []ID   `json:"media_ids"`
-	Sensitive   bool   `json:"sensitive"`
-	SpoilerText string `json:"spoiler_text"`
-	Visibility  string `json:"visibility"`
+	Status      string    `json:"status"`
+	InReplyToID ID        `json:"in_reply_to_id"`
+	MediaIDs    []ID      `json:"media_ids"`
+	Sensitive   bool      `json:"sensitive"`
+	SpoilerText string    `json:"spoiler_text"`
+	Visibility  string    `json:"visibility"`
+	ScheduledAt time.Time `json:"scheduled_at,omitempty"`
```

Comment:
Should be a pointer?

Reply:
I don't think anyone will ever run into a situation where they want to schedule a post for midnight on January 1st 0001

Reply:
If post the Toot without initialize ScheduledA, how the mastodon server handle?

Reply:
since it has `,omitempty`, Mastodon will not receive a `scheduled_at` field if `ScheduledAt` is the zero value of `time.Time`.

Reply:
Really? Did you try it?

https://play.golang.org/p/cvY3Bx0-Cpd

Reply:
Oh wow, I've been wrong all this time about what omitempty actually does

(It seems to have a hard-coded list of things it considers to be "zero")